prov/efa/src/rxr/rxr_pkt_type_data.c

```diff
@@ -231,10 +231,18 @@ void rxr_pkt_handle_data_send_completion(struct rxr_ep *ep,
 	struct rxr_tx_entry *tx_entry;
 
 	tx_entry = (struct rxr_tx_entry *)pkt_entry->x_entry;
-	tx_entry->bytes_acked +=
-		rxr_get_data_pkt(pkt_entry->pkt)->hdr.seg_size;
-
-	if (tx_entry->total_len == tx_entry->bytes_acked)
+	tx_entry->bytes_acked += rxr_get_data_pkt(pkt_entry->pkt)->hdr.seg_size;
+	if (tx_entry->total_len == tx_entry->bytes_acked &&
```

Comment:
this if condition is very complicated. Maybe break it into two if s?

Reply:
The branch prediction logic of `rxr_pkt_handle_data_send_completion` has been reimplemented.